src/snowflake/snowpark/session.py

```diff
@@ -2191,6 +2193,31 @@ def write_pandas(
                 str(ci_output)
             )
 
+    def _create_temp_table_for_given_schema(
+        self, temp_table_name: str, schema: StructType
+    ) -> bool:
+        """Creates a temp table for specified schema.
+
+        Args:
+            temp_table_name: table name
+            schema: user provided StructType schema
+
+        Returns:
+            True table was created successfully, else False
+        """
+        try:
+            schema_string = attribute_to_schema_string(schema._to_attributes())
+            self._run_query(
+                f"CREATE SCOPED TEMP TABLE {temp_table_name} ({schema_string})"
```

Comment:
I'm not particularly familiar with scoped temp tables. When do they get cleaned up if not explicitly deleted?

Reply:
When run outside of a stored proc, using `SCOPED` keyword is a no-op. Within stored proc, it is mainly used to limit the snowpark internal created temp object scope. This is not properly documented and I learnt about this by asking out in our slack channels